### DISCUSSION TITLE
how GeneActivity() function works

### DISCUSSION THREAD

**Discussion by lklindzi** [stuart-lab/signac#1180] [Q&A] [answered]
The new cellRanger-atac version 2.0.0 no longer provides the following metadata: enhancer_region_fragments, promoter_region_fragments, blacklist_region_fragments, DNase_sensitive_region_fragments. These metadata columns of seurat objects used to have values. But now, after running data through the new cellranger-atac pipeline, these columns are filled with zeros. Do any of Signac's functions rely on this metadata? I'm especially interested whether the GeneActivity() function relies on information from the 'promoter_region_fragments' metadata column.

**Comment by katimbach** [comment]
I also noticed after running counts with the new cellranger version (2.1.0) that all of my cells have 0 blacklist region fragments, which is usually used to filter cells as a QC step.

**Comment by timoast** [comment, marked as answer]
None of the cellranger metadata information is used when computing the gene activity scores, this comes from the gene annotations you use and the fragment file.

Blacklist region counts per cell can be computed using Signac as shown here: https://satijalab.org/signac/articles/pbmc_vignette.html#counting-fragments-in-genome-blacklist-regions
